src/messaging/Message.test.ts

```diff
@@ -61,62 +57,74 @@ import type {
+  let aliceAuthentication: SignerInterface<Signers.DidPalletSupportedAlgorithms, DidUrl>
   const aliceEncKey = makeEncryptionKeyTool('Alice//enc')
 
   let bobLightDid: DidDocument
   let bobLightDidWithDetails: DidDocument
   let bobFullDid: DidDocument
-  let bobSign: KeyToolSignCallback
+  let bobAuthentication: SignerInterface<Signers.DidPalletSupportedAlgorithms, DidUrl>
   const bobEncKey = makeEncryptionKeyTool('Bob//enc')
 
-  async function resolveKey(keyUri: DidResourceUri, keyRelationship = 'authentication'): Promise<ResolvedDidKey> {
-    const { did } = Did.parse(keyUri)
-    const document = [
-      aliceLightDidWithDetails,
-      aliceLightDid,
-      aliceFullDid,
-      bobLightDidWithDetails,
-      bobLightDid,
-      bobFullDid,
-    ].find(({ uri }) => uri === did)
-    if (!document) throw new Error('Cannot resolve mocked DID')
-    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    return Did.keyToResolvedKey(document[keyRelationship as keyof DidDocument]![0] as DidKey, did)
-  }
-
   beforeAll(async () => {
     await init()
-    const aliceAuthKey = makeSigningKeyTool('ed25519')
-    aliceSign = aliceAuthKey.getSignCallback
-    aliceLightDid = Did.createLightDidDocument({
+    const aliceAuthKey = await makeSigningKeyTool('ed25519')
+    aliceLightDid = createLightDidDocument({
       authentication: aliceAuthKey.authentication,
       keyAgreement: aliceEncKey.keyAgreement,
     })
-    aliceLightDidWithDetails = Did.createLightDidDocument({
+    aliceLightDidWithDetails = createLightDidDocument({
       authentication: aliceAuthKey.authentication,
       keyAgreement: aliceEncKey.keyAgreement,
       service: [{ id: '#id-1', type: ['type-1'], serviceEndpoint: ['x:url-1'] }],
     })
     aliceFullDid = await createLocalDemoFullDidFromLightDid(aliceLightDid)
+    aliceSign = await aliceAuthKey.getSigners(aliceFullDid)
+    aliceAuthentication = (
+      await aliceAuthKey.getSigners<Signers.DidPalletSupportedAlgorithms>(aliceFullDid, {
+        verificationRelationship: 'authentication',
+        algorithms: Signers.DID_PALLET_SUPPORTED_ALGORITHMS,
+      })
+    )[0]
 
-    const bobAuthKey = makeSigningKeyTool('ed25519')
-    bobSign = bobAuthKey.getSignCallback
-    bobLightDid = Did.createLightDidDocument({
+    const bobAuthKey = await makeSigningKeyTool('ed25519')
+    bobLightDid = createLightDidDocument({
       authentication: bobAuthKey.authentication,
       keyAgreement: bobEncKey.keyAgreement,
     })
-    bobLightDidWithDetails = Did.createLightDidDocument({
+    bobLightDidWithDetails = createLightDidDocument({
       authentication: bobAuthKey.authentication,
       keyAgreement: bobEncKey.keyAgreement,
       service: [{ id: '#id-1', type: ['type-1'], serviceEndpoint: ['x:url-1'] }],
     })
     bobFullDid = await createLocalDemoFullDidFromLightDid(bobLightDid)
+    bobAuthentication = (
+      await bobAuthKey.getSigners<Signers.DidPalletSupportedAlgorithms>(bobFullDid, {
```

Comment:
Do we need to have all the typing here? Also with the `aliceAuthentication`? It looks messy.

Reply:
It seems you made the helper `KeyToolSigners` later and not used it here. Please check if you have used it everywhere.